src/components/ContentSidebar/ActivitySidebar.js

```diff
@@ -56,9 +59,30 @@ export const activityFeedInlineError: Errors = {
         content: messages.activityFeedItemApiError,
     },
 };
+
+const MARK_NAME_JS_READY = `${ORIGIN_ACTIVITY_SIDEBAR}_${EVENT_JS_READY}`;
+
+window.performance.mark(MARK_NAME_JS_READY);
+
 class ActivitySidebar extends React.PureComponent<Props, State> {
+    static defaultProps = {
+        onMetric: noop,
+    };
+
     state = {};
 
+    constructor(props: Props) {
+        super(props);
+        this.props.onMetric(
+            METRIC_TYPE_ELEMENTS_LOAD_METRIC,
+            {
+                startMarkName: null, // TODO: replace with actual start mark once code splitting implemented
```

Comment:
Where will the new start mark come from? If it's from the parent component, should we instead pass down a callback to this component and allow the parent to handle decorating the metric payload with a start mark?

Reply:
The start mark will be in Sidebar.js. I can follow up with that in the code splitting PR since this data isn't really relevant at this point